engine/src/engine.rs

```diff
@@ -1032,6 +1034,10 @@ pub fn submit_with_alt_modexp<
         tx.try_into()
             .map_err(|_e| EngineErrorKind::InvalidSignature)?
     };
+    // Retrieve the signer of the transaction:
+    let sender = transaction.address;
+    // EIP-3706
+    check_empty_code(&io, &sender)?;
```

Comment:
Looks like here is not the best place for this check. Is it possible to do it at the EVM level?

Reply:
Definitely not. `SpuntikVM` operates as an execution layer. `EIP-3607` about transaction validation. It's same a ga  preparation in `Submit` call.
For example in REVM all the same - It's part of transaction validation before EVM execution.